Dockerfile

```diff
@@ -0,0 +1,36 @@
+ARG BASEIMAGE=rust:latest
+
+# BUILDER PATTERN
+
+FROM $BASEIMAGE AS builder
+
+WORKDIR /builder
+
+COPY . .
+
+RUN apt-get update
+RUN apt-get -y install libclang-dev libmetis-dev libscotch-dev
+
+ARG C_INCLUDE_PATH="/usr/include/x86_64-linux-musl"
+ARG CPLUS_INCLUDE_PATH="/usr/include/x86_64-linux-musl"
+
+RUN rustup target add x86_64-unknown-linux-musl
+RUN cargo build --workspace --release --bins --target x86_64-unknown-linux-musl --no-default-features
+
+# FINAL IMAGE
+
+FROM alpine
```

Comment:
Should be slim-bullseye as rust:latest is an alias to rust's bullseye image (see https://registry.hub.docker.com/_/rust/?tab=description)